### ISSUE TITLE
Config validation not done on templates

### ISSUE THREAD

**Issue by vosdev** [cloudalchemy/ansible-alertmanager#125]
**What happened?**

I made a mistake with my Go template. Ansible restarted Alertmanager. The mistake causes Alertmanager to crash on startup. There is no validation done.

**Did you expect to see some different?**
It would be nice to have a `amtool check-config` done before the restart.

Unfortunately it only accepts `alertmanager.yml` and not the `.tmpl` file directly, blocking the use of the `validate` option. Possible solution would be to have a task manually run amtool check-config, and restart if the templates were updated _and_ the check-config is OK.

**How to reproduce it (as minimally and precisely as possible)**:
Mess up a go template, run your playbook.

**Environment**

* Role version:

` 0.17.2`

* Ansible version information:

```
ansible 2.8.5
  config file = /var/lib/ansible/ansible.cfg
  configured module search path = ['/home/vos/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.7/dist-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 3.7.3 (default, Apr  3 2019, 05:39:12) [GCC 8.3.0]
```

* Variables:

```
insert role variables relevant to the issue
```

* Ansible playbook execution Logs:

```diff
TASK [cloudalchemy.alertmanager : copy alertmanager config] *******************************************************************************************************************************************************
ok: [prometheus1]

TASK [cloudalchemy.alertmanager : create systemd service unit] ****************************************************************************************************************************************************
ok: [prometheus1]

TASK [cloudalchemy.alertmanager : copy alertmanager template files] ***********************************************************************************************************************************************
--- before: /etc/alertmanager/templates/vos.tmpl
+++ after: /var/lib/ansible/playbooks/files/alertmanager/templates/vos.tmpl
@@ -1,10 +1,11 @@
+{{ invalidtemplatecode }}
 {{ define "slack.default.text" }}
     [PromQL Expression]({{ (index .Alerts 0).GeneratorURL }})
     {{ if gt (len .Alerts.Firing) 0 }}
         {{ range .Alerts.Firing }}{{ .Annotations.description }}
         {{ end }}
     {{ end }}
     {{ if gt (len .Alerts.Resolved) 0 }}
         **Resolved:**
         {{ range .Alerts.Resolved }}~~{{ .Annotations.description }}~~
         _start: {{ .StartsAt }}_

changed: [prometheus1] => (item=/var/lib/ansible/playbooks/files/alertmanager/templates/vos.tmpl)

TASK [cloudalchemy.alertmanager : ensure alertmanager service is started and enabled] *****************************************************************************************************************************
ok: [prometheus1]

RUNNING HANDLER [cloudalchemy.alertmanager : restart alertmanager] ************************************************************************************************************************************************
changed: [prometheus1]
```
```
May 10 22:18:41 prometheus1 systemd[1]: alertmanager.service: Main process exited, code=exited, status=1/FAILURE
May 10 22:18:41 prometheus1 systemd[1]: alertmanager.service: Failed with result 'exit-code'.
```

**Anything else we need to know?**:

**Comment by paulfantom** [comment]
> Unfortunately it only accepts alertmanager.yml and not the .tmpl file directly, blocking the use of the validate option. 

Since `validate` prevents copying incorrect file I would suggest creating an issue in alertmanager repo to allow running syntax checks on template files.

> Possible solution would be to have a task manually run amtool check-config, and restart if the templates were updated and the check-config is OK.

I don't see this fixing much. Simple server restart and you don't have working alertmanager as it won't start due to incorrect templates. This should be caught by failing `restart alertmanager` handler, but right now I have no idea why it didn't :(

**Comment by SuperQ** [comment]
**This role has been deprecated in favor of a the [prometheus-community/ansible](https://github.com/prometheus-community/ansible) collection.**
